lib/web_ui/lib/src/engine/safe_browser_api.dart

```diff
@@ -217,17 +217,18 @@ const bool _browserImageDecodingEnabled = bool.fromEnvironment(
 );
 
 /// Whether the current browser supports `ImageDecoder`.
-bool browserSupportsImageDecoder =
-  _browserImageDecodingEnabled &&
-  _imageDecoderConstructor != null &&
-  browserEngine == BrowserEngine.blink;
+bool browserSupportsImageDecoder = _defaultBrowserSupportsImageDecoder;
 
 /// Sets the value of [browserSupportsImageDecoder] to its default value.
 void debugResetBrowserSupportsImageDecoder() {
-  browserSupportsImageDecoder =
-      _imageDecoderConstructor != null;
+  browserSupportsImageDecoder = _defaultBrowserSupportsImageDecoder;
 }
 
+bool _defaultBrowserSupportsImageDecoder =
+    _browserImageDecodingEnabled &&
+    _imageDecoderConstructor != null &&
+    browserEngine == BrowserEngine.blink;
```

Comment:
@yjbanov I made this change to make sure we always reset it to the correct default (it looks like at some point things got out of sync).

Reply:
If it's not too much trouble, can you please move the `browserEngine == BrowserEngine.blink` into a constant with a TODO and a Github issue, something like this:

```dart
// TODO(yjbanov): $LINK_TO_GITHUB_ISSUE
// Frequently when a browser launches an API that other browsers already support
// there are subtle incompatibilities that may cause apps to crash if we blindly
// adopt the new implementation. This variable prevents us from picking up potentially
// incompatible implementations of ImagdeDecoder API. Instead, when a new browser
// engine launches the API, we'll evaluate it and enable it explicitly.
final bool _kProtectFromImageDecoderIncompatibilities = browserEngine == BrowserEngine.blink;
```

Reply:
I think it would be nice if we differentiated between `_browserImageDecodingEnabled` explicitly being set and defaulting to true. That way, if it is explicitly set, it can override the `browserEngine == BrowserEngine.blink`, so that it is easy to test against browsers that have implemented the ImageDecoder API.

Reply:
@eyebrowsoffire are you talking about overriding for tests or apps?

In tests, we already do override it e.g:
https://github.com/flutter/engine/blob/7a4542c76417441d899ab6b7f28ac9c00ef432dc/lib/web_ui/test/canvaskit/image_golden_test.dart#L64-L79

As for apps, do we really want users to play with this? Whoever wants to test this in different browsers likely already knows how to modify and build the engine, so they can just change the boolean value in the code.

Reply:
I meant in apps. If we wanted to test the viability of this in a browser, I'd much rather just be able to pass in a flag and try out a bunch of different apps than to rebuild the engine entirely. If a user stumbles upon the flag and misuses it, it's not the end of the world IMO. Let's just document this as well as we can, and if a user files a bug saying "I turned this flag on and it doesn't work" then we just say "don't do that" and close the bug and move on.